torch/distributed/_tensor/_utils.py

```diff
@@ -143,7 +143,11 @@ def compute_global_tensor_info(
     for idx, placement in enumerate(placements):
         mesh_dim_size = mesh.size(idx)
         if placement.is_shard():
-            shard_dim = cast(Shard, placement).dim
+            shard_placement = cast(Shard, placement)
+            if shard_placement.dim < 0:
+                shard_placement.dim += len(tensor_shape)
+            shard_dim = shard_placement.dim
+            # normalize shard dim to be positive
```

Comment:
nit: should the comment be a few lines above?

Reply:
good catch, done